lineage/lineage_graph_retrieval.py

```diff
@@ -0,0 +1,98 @@
+import pandas as pd
+from typing import List, Dict
+from pycarlo.core import Client, Query, Mutation
+import time
+
+
+def get_all_tables(client: Client, batch_size: int = 1000, selected_table_fields: List[str] = None, sleep_in_seconds: float = 0.5):
+    """
+    Retrieves all tables via getTable API: https://apidocs.getmontecarlo.com/#query-getTable
+    Tweak batch_size to increase throughput
+    Tweak sleep_in_seconds to stay within API limits
+    """
+    selected_table_fields = selected_table_fields or [
+        'mcon', 'project_name', 'dataset', 'table_id', 'table_type', 'full_table_id'
+    ]
+    current_cursor = None
+    tables = []
+
+    while (True):
+        params = {
+            'first': batch_size,
+            'is_deleted': False
+        }
+        if current_cursor:
+            params['after'] = current_cursor
+
+        query = Query()
+        get_tables_query = query.get_tables(**params)
+        get_tables_query.page_info()
+        get_tables_query.edges.node.__fields__(*selected_table_fields)
+
+        print(get_tables_query)
+
+        response = client(query)
+        for table in response.get_tables.edges:
+            tables.append({
+                field: table.node[field]
+                for field in selected_table_fields
+            })
+
+        has_next_page = response.get_tables.page_info.has_next_page
+        if not has_next_page:
+            break
+        current_cursor = response.get_tables.page_info.end_cursor
+
+        time.sleep(sleep_in_seconds)
```

Comment:
Why do we need the sleep?

Reply:
We have API limits (I forget exactly, but it limits X number of API requests per minute)

Reply:
so this is a hacky way of getting around it